### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by Ruofan1023**
update the following sections of DG:
- design: UI, Logic, Model and Storage implementation details
- manual testing guide

### CODE REVIEW

**Review by charliemoweng — COMMENTED** [review]
looks good 👍

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -125,17 +122,11 @@ How the parsing works:
 
 The `Model` component,
 
-* stores the address book data i.e., all `Person` objects (which are contained in a `UniquePersonList` object).
-* stores the currently 'selected' `Person` objects (e.g., results of a search query) as a separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
+* stores the teaching assistant buddy data i.e., all `Module` objects (which are contained in a `UniqueModuleList` object).
```
[the remark this answers is not in the document]
Comment (Ruofan1023) [reply docs/DeveloperGuide.md:125]: Yep agree will change!

**Review by tanjoewel — APPROVED** [review]
LGTM!

**Review by WangGLJoseph — APPROVED** [review]
LGTM！

**Pull request merged**